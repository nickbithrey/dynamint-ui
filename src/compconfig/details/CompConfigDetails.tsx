import * as React from 'react';
import { Form } from 'lib/form';
import { Field } from 'lib/form/field';
import { ICompConfig } from 'model/compconfig';
import Attributes, { Attribute } from './attributes';

export interface IDetails {
    id: string | number;
    item: ICompConfig;
    update: (item: ICompConfig) => void;
    back: () => void;
}

const Details = ({item, update, back}: IDetails) => {
    return (
        <Form entity={item} storageKey={'compConfigDetails'} save={update} back={back}>
            <Field
                type={'text'}
                name={'reference'}
                label={'Reference'}
                value={item.reference}
            />
            <Field
                type={'text'}
                name={'description'}
                label={'Description'}
                value={item.description}
            />
		    <Field
                type={'text'}
                name={'componentType'}
                label={'ComponentType'}
                value={item.componentType}
            />
            <Attributes link={item._links.attributes} entityForm={Attribute} />
        </Form>
    );
}

export default Details;