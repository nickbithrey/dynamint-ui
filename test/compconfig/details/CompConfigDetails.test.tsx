import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import CompConfigDetails, { IDetails } from '~/compconfig/details/CompConfigDetails';

describe('<CompConfigDetails />', () => {
    
    const props: IDetails = {
        id: 'id',
        item: {
            reference: 'val1',
            description: 'desc',
            componentType: 'type',
            _links: {
                attributes: {
                    href: 'href'
                }
            },
            uri: 'href'
        },
        update: jest.fn(),
        back: jest.fn()
    };

    let wrapper: ShallowWrapper;

    beforeEach(() => {
        wrapper = shallow(<CompConfigDetails {...props} />);
    });
    
    it('renders a valid tag and matches snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
});