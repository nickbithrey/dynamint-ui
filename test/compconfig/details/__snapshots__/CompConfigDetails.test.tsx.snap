// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CompConfigDetails /> renders a valid tag and matches snapshot 1`] = `
<Form
  back={[MockFunction]}
  entity={
    Object {
      "_links": Object {
        "attributes": Object {
          "href": "href",
        },
      },
      "componentType": "type",
      "description": "desc",
      "reference": "val1",
      "uri": "href",
    }
  }
  save={[MockFunction]}
  storageKey="compConfigDetails"
>
  <Component
    label="Reference"
    name="reference"
    type="text"
    value="val1"
  />
  <Component
    label="Description"
    name="description"
    type="text"
    value="desc"
  />
  <Component
    label="ComponentType"
    name="componentType"
    type="text"
    value="type"
  />
  <Connect(Loading)
    entityForm={[Function]}
    link={
      Object {
        "href": "href",
      }
    }
  />
</Form>
`;
